siliconcompiler/core.py

```diff
@@ -2764,15 +2764,15 @@ def check_logfile(self, jobname=None, step=None, index='0',
                     if string is not None:
                         matches[suffix] += 1
                         # always print to file
-                        print(string.strip(), file=checks[suffix]['report'])
+                        print(f'{num}: {string.strip()}', file=checks[suffix]['report'])
```

Comment:
I think just format the line once.
Also, I think it would be helpful if you right align the numbers (you can probably prefetch the total number of lines in the file to help with that:
```
  1: blah
 11: blah
111: blah blah
```